receiver/k8sclusterreceiver/internal/ingress/ingress.go

```diff
@@ -0,0 +1,104 @@
+package ingress
+
+import (
+	"fmt"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/maps"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/experimentalmetricmetadata"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	netv1 "k8s.io/api/networking/v1"
+	"strings"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+	imetadata "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/metadata"
+)
+
+const (
+	// Keys for ingress metadata.
+	AttributeK8SIngressUID  = "k8s.ingress.uid"
+	AttributeK8SIngressName = "k8s.ingress.name"
+	IngressCreationTime     = "ingress.creation_timestamp"
+)
+
+// Transform transforms the ingress to remove the fields.
+// IMPORTANT: Make sure to update this function before using new ingress fields.
+func Transform(r *netv1.Ingress) *netv1.Ingress {
+	newI := &netv1.Ingress{
+		ObjectMeta: metadata.TransformObjectMeta(r.ObjectMeta),
+	}
+	return newI
+}
+
+func RecordMetrics(mb *imetadata.MetricsBuilder, i *netv1.Ingress, ts pcommon.Timestamp) {
+	mb.RecordK8sIngressRuleCountDataPoint(ts, int64(len(i.Spec.Rules)))
+
+	rb := mb.NewResourceBuilder()
+	rb.SetK8sIngressUID(string(i.GetUID()))
+	rb.SetK8sIngressName(i.GetName())
+	rb.SetK8sClusterName("unknown")
```

Comment:
Why is this hardcoded to `"unknown"` ?

Reply:
Here, we are just adding the "k8s.cluster.name" key, the "unknown" is fallback value, 
when user will run the installation script the value will be filled

Reply:
![image](https://github.com/middleware-labs/opentelemetry-collector-contrib/assets/21999586/343e7f1f-a562-4137-b12e-ed942f831c0f)

Acc. to our code, "unknown" will be auto-replaced later.